org.eclipse.dawnsci.plotting.api/src/org/eclipse/dawnsci/plotting/api/tool/AbstractToolPage.java

```diff
@@ -496,13 +509,16 @@ protected IToolPage createToolInDedicatedView() throws Exception {
 	 * For instance write final datasets.
 	 */
 	public void exportFinished()  throws Exception {
-		
+		lazyWritables.clear();
 	}
 	
 	/**
 	 * Optionally override to do return an initial name for the NXData class saved at export 
 	 */
 	public String exportInit() {
+		exportIndex = 0;
+		if (lazyWritables == null)
+			lazyWritables = new HashMap<String, ILazyWriteableDataset>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)